command/agent/operator_endpoint.go

```diff
@@ -55,3 +56,110 @@ func (s *HTTPServer) OperatorRaftPeer(resp http.ResponseWriter, req *http.Reques
 	}
 	return nil, nil
 }
+
+// OperatorKeyringInstall is used to install a new gossip encryption key into the cluster
+func (s *HTTPServer) OperatorKeyringInstall(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "PUT" {
+		resp.WriteHeader(http.StatusMethodNotAllowed)
+		return nil, nil
+	}
+
+	var args structs.KeyringRequest
```

Comment:
This has extra fields that we wouldn't ever want to plumb directly (there's currently a `panic()` in the internal endpoint handler that we should change to an error if the Op isn't understood), so I'd make a simple unexported struct (define in here is fine) and a local string for the `token` and use those. There's stuff we don't use in this big struct so it's confusing to accept that here.